mods/beds/api.lua

```diff
@@ -46,6 +46,15 @@ function beds.register_bed(name, def)
 
 		on_place = function(itemstack, placer, pointed_thing)
 			local under = pointed_thing.under
+			local node = minetest.get_node(under)
+			local udef = minetest.registered_nodes[node.name]
+			if udef and udef.on_rightclick and
+					((not placer) or
+					(placer and not placer:get_player_control().sneak)) then
```

Comment:
`... and (not placer or not placer:get_player_control().sneak)` is shorter and has exactly the same effect.